test/hybridprov.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2025 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/crypto.h>
+#include <openssl/provider.h>
+#include <openssl/decoder.h>
+#include <openssl/encoder.h>
+#include <openssl/store.h>
+#include <openssl/rand.h>
+#include <openssl/core_names.h>
+#include "prov/provider_ctx.h"
+#include "testutil.h"
+
+extern const OSSL_DISPATCH ossl_mlx_kem_asym_kem_functions[];
+extern const OSSL_DISPATCH ossl_ml_kem_asym_kem_functions[];
+extern const OSSL_DISPATCH ossl_mlx_p256_kem_kmgmt_functions[];
+extern const OSSL_DISPATCH ossl_ml_kem_768_keymgmt_functions[];
+
+/* Defined in tls-provider.c */
+int hybrid_provider_init(const OSSL_CORE_HANDLE *handle,
+                         const OSSL_DISPATCH *in,
+                         const OSSL_DISPATCH **out,
+                         void **provctx);
+
+static const OSSL_ALGORITHM hybrid_asym_kem[] = {
+#ifndef OPENSSL_NO_ML_KEM
+    { "ML-KEM-768:MLKEM768:id-alg-ml-kem-768:2.16.840.1.101.3.4.4.2",
+      "provider=hybrid,fips=yes", ossl_ml_kem_asym_kem_functions },
+# if !defined(OPENSSL_NO_EC)
+    { "SecP256r1MLKEM768", "provider=hybrid,fips=yes", ossl_mlx_kem_asym_kem_functions },
+# endif
+#endif
+    { NULL, NULL, NULL }
+};
+
+#define PROV_NAMES_SecP256r1MLKEM768 "SecP256r1MLKEM768"
+#define PROV_DESCS_SecP256r1MLKEM768 "P-256+ML-KEM-768 TLS hybrid implementation"
+
+static const OSSL_ALGORITHM hybrid_keymgmt[] = {
+#ifndef OPENSSL_NO_ML_KEM
+    { "ML-KEM-768:MLKEM768:id-alg-ml-kem-768:2.16.840.1.101.3.4.4.2",
+      "provider=hybrid,fips=yes",
+      ossl_ml_kem_768_keymgmt_functions,
+      "OpenSSL ML-KEM-768 implementation" },
+# if !defined(OPENSSL_NO_EC)
+    { PROV_NAMES_SecP256r1MLKEM768, "provider=hybrid,fips=yes",
+      ossl_mlx_p256_kem_kmgmt_functions, PROV_DESCS_SecP256r1MLKEM768 },
+# endif
+#endif
+    { NULL, NULL, NULL }
+};
+
+static const OSSL_ALGORITHM *hybrid_query(void *provctx, int operation_id,
+                                          int *no_cache)
+{
+    *no_cache = 0;
+    switch (operation_id) {
+    case OSSL_OP_KEM:
+        return hybrid_asym_kem;
+    case OSSL_OP_KEYMGMT:
+        return hybrid_keymgmt;
```

Comment:
it may help to also include SHAKE here too, as internal only or fips=no. Such that base + fips + hybrid providers alone can achieve hybrid mlkem, without needing to load the default provider which has a high peak memory usage.